pkg/storage/cacher.go

```diff
@@ -497,14 +497,34 @@ func (c *cacheWatcher) stop() {
 	}
 }
 
+var timerPool sync.Pool
+
 func (c *cacheWatcher) add(event watchCacheEvent) {
-	t := time.NewTimer(5 * time.Second)
+	// Try to send the event immediately, without blocking.
+	select {
+	case c.input <- event:
+		return
+	default:
+	}
+
+	// OK, block sending, but only for up to 5 seconds.
+	// cacheWatcher.add is called very often, so arrange
+	// to reuse timers instead of constantly allocating.
+	const timeout = 5 * time.Second
+	t, ok := timerPool.Get().(*time.Timer)
+	if ok {
+		t.Reset(timeout)
```

Comment:
When retting a timer, don't we have to wait for the next event to drain?

https://github.com/golang/go/issues/11513

Pretty sure this can cause spurious ticks.